src/objects/Statistic.ts

```diff
@@ -73,18 +82,32 @@ export class Statistic<T, U> {
         this.statKey = key;
         this.statName = title;
         this.resetValue = resetValue;
+        this.minimumApiVersion = minimumApiVersion;
         this.subscriber = subscriber;
         this.accumulator = accumulator;
         this.formatDisplay = formatDisplay;
+    }
 
-        // Copilot says to bind the updateStat function to the class
-        // so that it can be passed as a callback
-        this.subscriber(this.updateStat.bind(this));
-
-        this.initialize();
+    /**
+     * Use to create a new statistic instead of the constructor.
+     * Handles checking if the API version is supported and initializes the stat.
+     */
+    static create<T, U>(...params: ConstructorParameters<typeof Statistic<T, U>>) {
+        const MIN_API_VERSION_ARG_INDEX = 3;
+        const minApiVersion = params[MIN_API_VERSION_ARG_INDEX];
+        if (context.apiVersion < minApiVersion) {
```

Comment:
Creating a constant here for `MIN_API_VERSION_ARG_INDEX ` feels like the most intentional and typesafe way to go about accessing minApiVersion, but I'm not totally satisfied long term. Now that there are 7 constructor args, I lean pretty strongly toward using an object with named keys for the params instead of having to index into the args array. It's not a dealbreaker for me, but it's on the edge of what I'm comfortable with.

Reply:
i could do that here in `create` without modifying the actual constructor, but i've opted to follow the pattern you established with using an args list instead of an object. let me know your preference one way or the other

Reply:
Fair honestly. It made a bit more sense before since the parameters were only in one place, but with this it makes more sense for an object. Especially since the parameter's index needs to be hardcoded, creating more complexity and potentially causing problems if the constructor changes again.

Reply:
i made this change in my latest push. I'm satisfied with all changes now; happy to update contribution doc and call it a wrap.

Reply:
Sounds good. Don't think much needs to be added there other than mentioning the `minimumApiVersion` and how `0` would be the indicator value that it works for all supported versions.

Reply:
I don't love what I wrote but I think it does the job. Happy for feedback.